posthog/hogql_queries/test/test_lifecycle_hogql_query.py

```diff
@@ -0,0 +1,161 @@
+
+        # TODO probably doesn't make sense to test like this
+        #  maybe this query should be what is returned by the function
+        events_query = create_events_query("day", event_filter=time_filter)
+        return execute_hogql_query(
+            team=self.team,
+            query="""
+                    SELECT
+                    start_of_period, count(DISTINCT person_id) AS counts, status
+                    FROM {events_query}
+                    GROUP BY start_of_period, status
+                    """,
+            query_type="LifecycleQuery",
+            placeholders={"events_query": events_query},
+        )
+
+    def test_events_query_whole_range(self):
+        self._create_test_events()
+
+        date_from = "2020-01-09"
+        date_to = "2020-01-19"
+
+        response = self.run_events_query(date_from, date_to, IntervalType.day)
+
+        self.assertEqual(
+            {
+                (datetime(2020, 1, 9, 0, 0), 1, "new"),  # p2
+                (datetime(2020, 1, 10, 0, 0), 1, "dormant"),  # p2
+                (datetime(2020, 1, 11, 0, 0), 1, "new"),  # p1
+                (datetime(2020, 1, 12, 0, 0), 1, "new"),  # p3
+                (datetime(2020, 1, 12, 0, 0), 1, "resurrecting"),  # p2
+                (datetime(2020, 1, 12, 0, 0), 1, "returning"),  # p1
+                (datetime(2020, 1, 13, 0, 0), 1, "returning"),  # p1
+                (datetime(2020, 1, 13, 0, 0), 2, "dormant"),  # p2, p3
+                (datetime(2020, 1, 14, 0, 0), 1, "dormant"),  # p1
+                (datetime(2020, 1, 15, 0, 0), 1, "resurrecting"),  # p1
+                (datetime(2020, 1, 15, 0, 0), 1, "new"),  # p4
+                (datetime(2020, 1, 16, 0, 0), 2, "dormant"),  # p1, p4
+                (datetime(2020, 1, 17, 0, 0), 1, "resurrecting"),  # p1
+                (datetime(2020, 1, 18, 0, 0), 1, "dormant"),  # p1
+                (datetime(2020, 1, 19, 0, 0), 1, "resurrecting"),  # p1
+                (datetime(2020, 1, 20, 0, 0), 1, "dormant"),  # p1
+            },
+            set(response.results),
+        )
+
+    def test_events_query_partial_range(self):
+        self._create_test_events()
+        date_from = "2020-01-12"
+        date_to = "2020-01-14"
+        response = self.run_events_query(date_from, date_to, IntervalType.day)
+
+        self.assertEqual(
+            {
+                (datetime(2020, 1, 11, 0, 0), 1, "new"),  # p1
+                (datetime(2020, 1, 12, 0, 0), 1, "new"),  # p3
+                (datetime(2020, 1, 12, 0, 0), 1, "resurrecting"),  # p2
+                (datetime(2020, 1, 12, 0, 0), 1, "returning"),  # p1
+                (datetime(2020, 1, 13, 0, 0), 1, "returning"),  # p1
+                (datetime(2020, 1, 13, 0, 0), 2, "dormant"),  # p2, p3
+                (datetime(2020, 1, 14, 0, 0), 1, "dormant"),  # p1
+            },
+            set(response.results),
+        )
+
+    # def test_start_on_dormant(self):
```

Comment:
This seems to be missing tests for the lifecycle query itself.

Reply:
Would still like to see at least one test for lifecycles, if only "this query doesn't error".